themes/src/main/resources/theme/keycloak-preview/account/index.ftl

```diff
@@ -121,6 +122,23 @@
             }).error(function() {
                 alert('failed to initialize keycloak');
             });
+
+            function loggedInUserName() {
+                let userName = '${msg("unknownUser")}';
+                if (keycloak.tokenParsed) {
+                    const givenName = keycloak.tokenParsed.given_name;
+                    const familyName = keycloak.tokenParsed.family_name;
+                    const preferredUsername = keycloak.tokenParsed.preferred_username;
+                    if (givenName && familyName) {
+                        userName = [givenName, familyName].reduce((acc, value, index) =>
+                            acc.replace('{'+ index + '}', value), '${msg("fullName")}'
+                        );
+                    } else {
+                        userName = (givenName || familyName) || preferredUsername || userName;
+                    }
+                }
+                return userName;
+            }
```

Comment:
@edewit Can you move this function to WelcomePageScripts.js?  index.ftl is too big and hard to manage already.

Reply:
It can't move as it needs freemarker to have the "`fullName = {0} {1}`" template